vendor/bat-native-ledger/src/bat/ledger/internal/bat_get_media.cc

```diff
@@ -41,44 +42,44 @@ std::string BatGetMedia::GetLinkType(const std::string& url,
 
 void BatGetMedia::ProcessMedia(const std::map<std::string, std::string>& parts,
                                const std::string& type,
-                               const ledger::VisitData& visit_data) {
-  if (parts.size() == 0 || !ledger_->GetRewardsMainEnabled()) {
+                               ledger::VisitDataPtr visit_data) {
+  if (parts.size() == 0 || !ledger_->GetRewardsMainEnabled() || !visit_data) {
     return;
   }
 
   if (type == YOUTUBE_MEDIA_TYPE) {
-    media_youtube_->ProcessMedia(parts, visit_data);
+    media_youtube_->ProcessMedia(parts, *visit_data);
```

Comment:
Why wouldn't we use pointer at this point as well? Why conversion?

Reply:
the way these to `if` statements are structured with return statements is very strange. If we had an if/else it would be much safer to std::move the visit_data without worrying about someone changing it in a way that attempted to access it after move